packages/db/test/pinning.spec.js

```diff
@@ -0,0 +1,185 @@
+  assert.ok(typeof pinRequestOutput._id === 'string', '_id should be a string')
+  assert.ok(typeof pinRequestOutput.requestedCid === 'string', 'requestedCid should be a string')
+  assert.ok(Array.isArray(pinRequestOutput.pins), 'pin should be an array')
+  assert.ok(Date.parse(pinRequestOutput.created), 'created should be valid date string')
+  assert.ok(Date.parse(pinRequestOutput.updated), 'updated should be valid date string')
+
+  if (withContent) {
+    assert.ok(typeof pinRequestOutput.contentCid === 'string', 'requestedCid should be a string')
+  } else {
+    assert.ifError(pinRequestOutput.contentCid)
+  }
+}
+
+describe('Pin Request', () => {
+  /** @type {DBClient & {_client}} */
+  const client = (new DBClient({
+    endpoint: 'http://127.0.0.1:3000',
+    token
+  }))
+  let user
+  let authKey
+  /**
+   * @type {import('../db-client-types').PAPinRequestUpsertInput}
+   */
+  let aPinRequestInput
+
+  /**
+   * @type {import('../db-client-types').PAPinRequestUpsertOutput}
+   */
+  let aPinRequestOutput
+
+  /**
+   * @type {import('../db-client-types').PAPinRequestUpsertInput}
+   */
+  let aPinRequestInputForExistingContent
+
+  /**
+   * @type {import('../db-client-types').PAPinRequestUpsertOutput}
+   */
+  let aPinRequestOutputForExistingContent
+
+  const cids = [
+    'bafybeiczsscdsbs7aaqz55asqdf3smv6klcw3gofszvwlyarci47bgf356',
+    'bafybeiczsscdsbs7aaqz55asqdf3smv6klcw3gofszvwlyarci47bgf358'
+  ]
+
+  const pins = [
+    {
+      status: 'Pinning',
+      location: {
+        peerId: '12D3KooWFe387JFDpgNEVCP5ARut7gRkX7YuJCXMStpkq714ziK6',
+        peerName: 'web3-storage-sv15',
+        region: 'region'
+      }
+    },
+    {
+      status: 'Pinning',
+      location: {
+        peerId: '12D3KooWFe387JFDpgNEVCP5ARut7gRkX7YuJCXMStpkq714ziK7',
+        peerName: 'web3-storage-sv16',
+        region: 'region'
+      }
+    }
+  ]
+
+  // Create user and auth key`
+  before(async () => {
+    user = await createUser(client)
+    authKey = await createUserAuthKey(client, user._id)
+  })
+
+  // Guarantee no Pin requests exist and create the ones needed for our tests
+  before(async () => {
+    // Make sure we don't have pinRequest and content
+    await client._client.from(pinRequestTable).delete()
+    const { count: countR } = await client._client.from(pinRequestTable).select('id', {
+      count: 'exact'
+    })
+    assert.strictEqual(countR, 0, 'There are still requests in the db')
+
+    await createUpload(client, user._id, authKey, cids[1], { pins: pins })
+
+    aPinRequestInput = {
+      requestedCid: cids[0],
+      authKey
+    }
+
+    aPinRequestInputForExistingContent = {
+      requestedCid: cids[1],
+      authKey
+    }
+
+    aPinRequestOutput = await client.createPAPinRequest(aPinRequestInput)
+    aPinRequestOutputForExistingContent = await client.createPAPinRequest(aPinRequestInputForExistingContent)
+  })
+
+  describe('Create Pin', () => {
+    it('it creates a Pin Request', async () => {
```

Comment:
Double "it" - can we remove in all instances 🙏?

```suggestion
    it('creates a Pin Request', async () => {
```

Reply:
Done in #810